### PR TITLE
chore: include all supported connectors in the create fuels template

### LINKED ISSUE #2465: `create fuels` preceived slowness
- Validate if this is from our side with from excessive re-rendering
- If not, investigate further with the wallet SDK

### LINKED ISSUE #2677: Include all supported connectors in the `create fuels` template
Right now the `create fuels` template only includes one connector - the Fuel Wallet. We should include all the supported ones: https://github.com/FuelLabs/fuel-connectors/tree/main/packages

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- Closes #2677 
- Closes #2465

# Release notes

- Added support for multiple new connectors in the `create fuels` template

# Summary

This PR bumps the `@fuels/connectors` package to `0.8.1` which includes all the new connectors.

image

This PR also fixes the 'slowness' associated with the 'Connect Wallet' button.

# Checklist

- [x] I **_added_** — `tests` to prove my changes
- [x] I **_updated_** — all the necessary `docs`
- [x] I **_reviewed_** — the entire PR myself, using the GitHub UI
- [x] I **_described_** — all breaking changes and the Migration Guide

### CODE REVIEW

**Review by Torres-ssf — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Review by maschad — APPROVED** [review]

**Comment by Dhaiwat10** [comment]
Marking this as a draft as it seems like there is some additional configuration that needs to be done in order for these new connectors to work on localhost _and_ testnet

**Comment by Dhaiwat10** [comment]
Should be good to go now ✅

File: apps/create-fuels-counter-guide/sway-programs/predicate/src/main.sw [hunk 1]
```diff
@@ -1,9 +1,5 @@
 predicate;
 
-configurable {
-    PIN: u64 = 1337,
```
Comment (Dhaiwat10) [inline apps/create-fuels-counter-guide/sway-programs/predicate/src/main.sw:4]: Looks like an unrelated change crept in. Will fix this

File: .changeset/tall-rockets-hide.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
Comment (arboleya) [inline .changeset/tall-rockets-hide.md:2]: Empty?

Comment (Dhaiwat10) [reply .changeset/tall-rockets-hide.md:2]: Oops good catch - fixed. https://github.com/FuelLabs/fuels-ts/pull/2691/commits/ba09ad9a805a8db315b68f4f9acbd9b0e9ebb1fc

File: apps/create-fuels-counter-guide/src/components/Link.tsx
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/components/Link.tsx:1]: I have just copied the source code from the template as it is. Looks like this was missed in the PR that closed that issue. Doesn't seem to be related to this PR

File: apps/create-fuels-counter-guide/sway-programs/fuel-toolchain.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[toolchain]
+channel = "testnet"
+
+[components]
+forc = "0.61.2"
+fuel-core = "0.30.0"
```
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/sway-programs/fuel-toolchain.toml:6]: Good idea about the sync automation, but it should probably be in the form of a CI check that ensures the versions are always aligned.

**Comment by Dhaiwat10** [comment]
@Torres-ssf apparently we were incorrectly importing/exporting the `isBrowserWalletConnected` state from the `useBrowserWallet` hook and that was causing the status to change slowly or not change at all.

File: apps/create-fuels-counter-guide/src/components/Link.tsx
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/components/Link.tsx:1]: @petertonysmith94 should I re-open the issue and address it in a separate PR then? This PR already closes two issues

File: .changeset/tall-rockets-hide.md (hunk 2 above)
Comment (danielbate) [reply .changeset/tall-rockets-hide.md:2]: Does this not need a `create-fuels` bump?

File: apps/create-fuels-counter-guide/src/app/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                  new FuelWalletConnector(),
+                  new BurnerWalletConnector({
+                    fuelProvider: providerToUse,
+                  }),
+                  new WalletConnectConnector({
+                    fuelProvider: providerToUse,
+                  }),
+                  new BakoSafeConnector(),
+                  new FueletWalletConnector(),
+                  new FuelWalletDevelopmentConnector(),
```
Comment (danielbate) [inline apps/create-fuels-counter-guide/src/app/layout.tsx:53]:
```suggestion
```

The dev wallet connector doesn't seem to be a [default connector](https://github.com/FuelLabs/fuel-connectors/blob/v0.8.1/packages/connectors/src/defaultConnectors.ts#L17-L21)?

File: apps/create-fuels-counter-guide/sway-programs/fuel-toolchain.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielbate) [reply apps/create-fuels-counter-guide/sway-programs/fuel-toolchain.toml:6]: > it should probably be in the form of a CI check that ensures the versions are always aligned

There is a test that will fail if they go out of alignment.

But yes we could definitely improve/automate more areas where we manually change the versions.

File: templates/nextjs/src/app/layout.tsx [hunk 5]
```diff
@@ -17,22 +24,36 @@ interface RootLayoutProps {
...
+                  new FuelWalletConnector(),
+                  new BurnerWalletConnector({
+                    fuelProvider: providerToUse,
+                  }),
+                  new WalletConnectConnector({
+                    fuelProvider: providerToUse,
+                  }),
+                  new BakoSafeConnector(),
+                  new FueletWalletConnector(),
+                  new FuelWalletDevelopmentConnector(),
```
Comment (danielbate) [inline templates/nextjs/src/app/layout.tsx:53]: As above, re usage of the dev wallet connector.

File: apps/create-fuels-counter-guide/sway-programs/fuel-toolchain.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/sway-programs/fuel-toolchain.toml:6]: There is a test already? @danielbate

File: apps/create-fuels-counter-guide/src/app/layout.tsx (hunk 4 above)
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/app/layout.tsx:53]: I thought we should include all of them instead of only the default ones and let the user remove the ones that they don't need. Easier to remove than to having to find and import something that isn't there. Thoughts?

File: .changeset/tall-rockets-hide.md (hunk 2 above)
Comment (Dhaiwat10) [reply .changeset/tall-rockets-hide.md]: https://github.com/FuelLabs/fuels-ts/pull/2691/commits/1dda01396dc321f0f4684ad5d0d2156c9ddb636d

Fixed

File: apps/create-fuels-counter-guide/src/components/Link.tsx
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/components/Link.tsx:1]: Re-opened the issue

**Review by petertonysmith94 — COMMENTED** [review]
Looks good to me - just need to upgrade the fuel-core versions

**Review by petertonysmith94 — DISMISSED** [review]
❇️

File: apps/create-fuels-counter-guide/src/app/layout.tsx (hunk 4 above)
Comment (danielbate) [reply apps/create-fuels-counter-guide/src/app/layout.tsx:53]: Makes sense, just wanted to confirm 🫡

**Review by danielbate — APPROVED** [review]

**Review by petertonysmith94 — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Review by maschad — APPROVED** [review]
